packages/prompts/src/autocomplete.ts

```diff
@@ -0,0 +1,298 @@
+import { AutocompletePrompt } from '@clack/core';
+import color from 'picocolors';
+import {
+	type CommonOptions,
+	S_BAR,
+	S_BAR_END,
+	S_CHECKBOX_INACTIVE,
+	S_CHECKBOX_SELECTED,
+	S_RADIO_ACTIVE,
+	S_RADIO_INACTIVE,
+	symbol,
+} from './common.js';
+import { limitOptions } from './limit-options.js';
+import type { Option } from './select.js';
+
+function getLabel<T>(option: Option<T>) {
+	return option.label ?? String(option.value ?? '');
+}
+
+function getFilteredOption<T>(searchText: string, option: Option<T>): boolean {
+	if (!searchText) {
+		return true;
+	}
+	const label = (option.label ?? String(option.value ?? '')).toLowerCase();
+	const hint = (option.hint ?? '').toLowerCase();
+	const value = String(option.value).toLowerCase();
+	const term = searchText.toLowerCase();
+
+	return label.includes(term) || hint.includes(term) || value.includes(term);
+}
+
+function getSelectedOptions<T>(values: T[], options: Option<T>[]): Option<T>[] {
+	const results: Option<T>[] = [];
+
+	for (const option of options) {
+		if (values.includes(option.value)) {
+			results.push(option);
+		}
+	}
+
+	return results;
+}
+
+export interface AutocompleteOptions<Value> extends CommonOptions {
+	/**
+	 * The message to display to the user.
+	 */
+	message: string;
+	/**
+	 * Available options for the autocomplete prompt.
+	 */
+	options: Option<Value>[];
+	/**
+	 * The initial selected value.
+	 */
+	initialValue?: Value;
+	/**
+	 * Maximum number of items to display at once.
+	 */
+	maxItems?: number;
+	/**
+	 * Placeholder text to display when no input is provided.
+	 */
+	placeholder?: string;
+}
+
+export const autocomplete = <Value>(opts: AutocompleteOptions<Value>) => {
+	const prompt = new AutocompletePrompt<Option<Value>>({
+		options: opts.options,
+		placeholder: opts.placeholder,
+		initialValue: opts.initialValue ? [{ value: opts.initialValue }] : undefined,
```

Comment:
this was right before

the `initialValue` is the value, not the option. it was meant to be `[opts.initialValue]`

presumably this is also why you added `Option<Value>`, because it was inferring the right type before but adding this broke it (correctly)